Docs: Azure AD/ Entra ID docs improvements (#96225)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md

1. Click **Users and Groups**.
1. Click **Add user/group** to add a user or group to the Grafana roles.

{{% admonition type="note" %}}

> If you assign a group to a Grafana role, ensure that users are direct members of the group. Users in nested groups will not have access to Grafana due to a limitation on the Azure AD/Entra ID side. For more information, see [Microsoft Entra service limits and restrictions](https://learn.microsoft.com/en-us/entra/identity/users/directory-service-limits-restrictions).

Reviewer: Just a nit again.
```suggestion
When assigning a group to a Grafana role, ensure that users are direct members of the group. Users in nested groups will not have access to Grafana due to limitations within Azure AD/Entra ID side. For more information, see [Microsoft Entra service limits and restrictions](https://learn.microsoft.com/en-us/entra/identity/users/directory-service-limits-restrictions).
```

# prevents the sync of org roles from AzureAD
skip_org_role_sync = true
```

## Configuration options

> The following table outlines the various generic Azure AD/Entra ID configuration options. You can apply these options as environment variables, similar to any other configuration within Grafana.

Reviewer: nit: maybe add a link to https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#override-configuration-with-environment-variables ?

Reviewer: Hey @mgyongyosi sorry for extra comment - I am working on another issue where it mentions it is unclear how to avoid any errors when the JMESPath expression includes a colon. As we don't have one single place talking about this - what do you think about adding something around the lines of:

```
Note: If the configuration option requires a [JMESPath](http://jmespath.org/examples.html) expression that includes a colon (:), enclose the entire expression in quotes to prevent parsing errors. For example:

role_attribute_path: "role:view"

```

Author: I would prefer to have a separate page for JMESPath examples, instead of adding the same/similar example to all of the pages, wdyt?

Reviewer: > I would prefer to have a separate page for JMESPath examples, instead of adding the same/similar example to all of the pages, wdyt?

I agree, let's review together what would be the best way to structure those.

Reviewer: FYI bunch of other places where I had to add https://github.com/grafana/grafana/pull/96237/files (not suggesting to merge mine, as we talked above, let's see if we can extract this info to some helper content)

| `auto_login`                 | No       | Yes                | Set to `true` to enable users to bypass the login screen and automatically log in. This setting is ignored if you configure multiple auth providers to use auto-login.                                                                                                                                                                                                                                                                                                                                                                                                                                     | `false`                |

> | `role_attribute_path`        | No       | Yes                | [JMESPath](http://jmespath.org/examples.html) expression to use for Grafana role lookup. Grafana will first evaluate the expression using the OAuth2 ID token. If no role is found, the expression will be evaluated using the user information obtained from the UserInfo endpoint. The result of the evaluation should be a valid Grafana role (`None`, `Viewer`, `Editor`, `Admin` or `GrafanaAdmin`). For more information on user role mapping, refer to [Configure role mapping]({{< relref "#configure-role-mapping" >}}).                                                                          |                        |

Reviewer: I see `#map-roles` header in the page, but not `#configure-role-mapping`.

Author: Done

> | `role_attribute_strict`      | No       | Yes                | Set to `true` to deny user login if the Grafana org role cannot be extracted using `role_attribute_path` or `org_mapping`. For more information on user role mapping, refer to [Configure role mapping]({{< relref "#configure-role-mapping" >}}).                                                                                                                                                                                                                                                                                                                                                         | `false`                |

Reviewer: Here as well the rel link seems to be incorrect.

Author: Done, updated to the correct one.

Author: Done
